Intro_to_R_Tidyverse_cheatsheet (#139)
AlexsLemonade/training-modules: module_cheatsheets/Intro_to_R_Tidyverse_cheatsheet.md

| Base `R`          | [`==, <=, >=, !=`](https://stat.ethz.ch/R-manual/R-devel/library/base/html/Comparison.html)       | Relational Operators  | These are binary operators which allow for the comparison of values in an object.                                                                                                                 |
| Base `R`          | [`str(x)`](https://www.rdocumentation.org/packages/utils/versions/3.5.1/topics/str)               | Object Structure      | Gets a summary of the object `x` structure.                                                                                                                                                         |
| Base `R`          | [`head(x)`](https://www.rdocumentation.org/packages/utils/versions/3.5.1/topics/head)             | Head                  | Returns the top 6 rows of an object in the environment by default. You can specify how many rows you want by including the `n = `argument.                                                        |

> | Base `R`          | [`tail(x)`](https://www.rdocumentation.org/packages/utils/versions/3.5.1/topics/head)             | Tail                  | Returns the bottom 6 rows of an object in the environment. You can specify how many rows you want by including the `n =` argument.                                                                  |

Reviewer: ```suggestion
| Base `R`          | [`tail()`](https://www.rdocumentation.org/packages/utils/versions/3.5.1/topics/head)             | Tail                  | Returns the bottom 6 rows of an object in the environment by default. You can specify how many rows you want by including the `n =` argument.                                                                  |
```

The addition of `by default` makes it consistent with the line above this. I would remove the `x` here and anywhere else you do not explicitly talk about it in the last column.

> | Base `R`          | [`as.factor(x)`](https://www.rdocumentation.org/packages/base/versions/3.5.1/topics/factor)       | As Factor             | Coerces object x into a factor (which is used to represent categorical data). This function can be used to coerce object `x` into other data types i.e.. as.character, as.data.frame, as.matrix, etc. |

Reviewer: ```suggestion
| Base `R`          | [`as.factor(x)`](https://www.rdocumentation.org/packages/base/versions/3.5.1/topics/factor)       | As Factor             | Coerces object x into a factor (which is used to represent categorical data). This function can be used to coerce object `x` into other data types, i.e., `as.character`, `as.data.frame`, `as.matrix`, etc. |
```

| Base `R`          | [`dim(x)`](https://www.rdocumentation.org/packages/base/versions/3.5.1/topics/dim)                | Dimensions summary    | Returns the dimensions of object `x`.                                                                                                                                                               |
| Base `R`          | [`data.frame()`](https://www.rdocumentation.org/packages/base/versions/3.5.1/topics/data.frame)   | Data Frame            | Creates a data.frame where the named arguments will be the same length                                                                                             |

> | Base `R`          | `[[x]]`[click here](https://www.rdocumentation.org/packages/base/versions/3.5.1/topics/Extract)             |  Double bracket       | Extracts `x` from a nested list                                |

Reviewer: ```suggestion
| Base `R`          | [`[[x]]`](https://www.rdocumentation.org/packages/base/versions/3.5.1/topics/Extract)             |  Double bracket       | Extracts `x` from a nested list                                |
```

Author: I tried this format and it disrupted the format of the table (the information from the following two columns becomes squashed into this column). Any thoughts on that?

Reviewer: Don't know _why_ but I am pretty sure that a switch to HTML format will do it:
```suggestion
| Base `R`          | <a href = "https://www.rdocumentation.org/packages/base/versions/3.5.1/topics/Extract">`[[x]]`</a>             |  Double bracket       | Extracts `x` from a nested list                                |
```

Author: Great, that works

| dplyr           | [`%>%`](https://www.rdocumentation.org/packages/magrittr/versions/1.5/topics/%25%26gt%3B%25)        | Pipe operator         | Funnels a data.frame through tidyverse operations                                                                                                                                                 |
| dplyr           | [`filter()`](https://www.rdocumentation.org/packages/dplyr/versions/0.7.8/topics/filter)            | Filter                | Returns a subset of rows matching the conditions of the specified logical argument                                                                                                                |
| dplyr           | [`arrange()`](https://www.rdocumentation.org/packages/dplyr/versions/0.7.8/topics/arrange)          | Arrange               | Reorders rows in ascending order. arrange(desc()) would reorder rows in descending order.                                                                                                         |

> | dplyr           | [`select()`](https://www.rdocumentation.org/packages/dplyr/versions/0.7.8/topics/select)            | Select                | Isolates columns that match the specified argument                                                                                                                                                |

Reviewer: Would _selects_ be a better word than _isolates_?

| dplyr           | [`mutate()`](https://www.rdocumentation.org/packages/dplyr/versions/0.7.8/topics/mutate)            | Mutate                | Adds a new column that is a function of existing columns                                                                                                                                          |

> | dplyr           | [`summarise()`](https://www.rdocumentation.org/packages/dplyr/versions/0.7.8/topics/summarise)      | Summarise             | Summarises multiple values in an object into a single value. This function can be used with other functions to retrieve a single output value for the grouped values.                                                                                                     |

Reviewer: May want to also add `summarize` here -- I've had people ask about the spelling and if it is equivalent before.

### `readr` and `tibble`
Read the `readr` package documentation [**here**](https://www.rdocumentation.org/packages/readr/versions/1.3.0) and the package vignette [**here**](https://www.rdocumentation.org/packages/readr/versions/1.3.0/vignettes/readr.Rmd). <br>
Read the `tibble` package documentation [**here**](https://www.rdocumentation.org/packages/tibble/versions/1.4.2) and the package vignette [**here**](https://www.rdocumentation.org/packages/tibble/versions/1.4.2/vignettes/tibble.Rmd).

| Library/Package                      | Piece of code                                   | What it's called       | What it does                                                                                                                                                                                      |
|--------------------------------------|-----------------------------------------------------|---------------------------|-----------------------------------------------------------------------------------------------------------------------------------|

> | readr           | [`read_tsv()`](https://www.rdocumentation.org/packages/readr/versions/1.3.0/topics/read_delim)           | Read .TSV             | Reads in a .TSV file from a specified file path. This function can be tailored to read in other common types of files. I.e. read_csv(), read_rds(), etc.                                          |

Reviewer: ```suggestion
| readr           | [`read_tsv()`](https://www.rdocumentation.org/packages/readr/versions/1.3.0/topics/read_delim)           | Read TSV             | Reads in a TSV file from a specified file path. This function can be tailored to read in other common types of files, i.e. read_csv(), read_rds(), etc.                                          |
```
I would do `.tsv` when describing the extension and TSV when talking about the file.
